spark/common/src/test/scala/org/apache/sedona/sql/rasteralgebraTest.scala

```diff
@@ -1296,6 +1296,58 @@ class rasteralgebraTest extends TestBaseScala with BeforeAndAfter with GivenWhen
       assertEquals("UNSIGNED_8BITS", result)
     }
 
+    it("Passed RS_MapAlgebra with two raster columns") {
+      val df1 = sparkSession.read.format("binaryFile")
+        .option("recursiveFileLookup", "true")
+        .option("pathGlobFilter", "*.tif*")
+        .load(resourceFolder + "raster")
+        .selectExpr("path", "RS_FromGeoTiff(content) as rast0")
+      val df2 = sparkSession.read.format("binaryFile")
+        .option("recursiveFileLookup", "true")
+        .option("pathGlobFilter", "*.tif*")
+        .load(resourceFolder + "raster")
+        .selectExpr("path", "RS_FromGeoTiff(content) as rast1")
+        .withColumnRenamed("path", "path1")
+      val df = df1.join(df2, df1("path") === df2("path1"), "inner").select("rast0", "rast1", "path")
+      Seq(null, "b", "s", "i", "f", "d").foreach { pixelType =>
+        val pixelTypeExpr = if (pixelType == null) null else s"'$pixelType'"
+        val dfResult = df.withColumn("rast_2", expr(s"RS_MapAlgebra(rast0, rast1, $pixelTypeExpr, 'out[0] = rast0[0] * 0.5 + rast1[0] * 0.5;', null)"))
```

Comment:
Can the script have `out[0]`? What is the difference between `out` and `out[0]`?

Reply:
Yes, and it doesn't matter as we are return a single band raster.